libraries/chain/controller.cpp

```diff
@@ -1013,6 +1014,19 @@ struct controller_impl {
       );
    }
 
+   // search on the branch of given id
+   block_state_ptr fork_db_fetch_bsp_by_num(const block_id_type& id, uint32_t block_num) const {
+      return fork_db.apply<block_state_ptr>(
+         overloaded{
+            [](const fork_database_legacy_t&) -> block_state_ptr { return nullptr; },
```

Comment:
I suppose we don't use it. But we could do a `search_on_branch` with the legacy forkdb as well.

Reply:
Not that returns the same type.

Reply:
Oh right. Never mind then.